### PR TITLE
✨ Unit test for generateFlags

### PR DESCRIPTION
**Pull request by laurentsimon**
see title

### CODE REVIEW

**Comment by laurentsimon** [comment]
Any idea why the CI workflow is not being run?

**Comment by asraa** [comment]
> Any idea why the CI workflow is not being run?

I wonder if it's a YAML error, lemme try test parsing it

**Comment by laurentsimon** [comment]
https://github.com/asraa/slsa-on-github/pull/42

File: build-go/pkg/build_test.go [hunk 1]
```diff
@@ -686,3 +686,77 @@ func TestGenerateLdflags(t *testing.T) {
...
+			fmt.Println(err)
+			if !errCmp(err, tt.expected) {
+				t.Errorf(cmp.Diff(err, tt.expected))
+			}
+			if err != nil {
+				return
+			}
+			// Note: generated env variables contain the process's env variables too.
+			if !cmp.Equal(flags, expectedFlags) {
+				t.Errorf(cmp.Diff(flags, expectedFlags))
```
Comment (asraa) [inline build-go/pkg/build_test.go:758]: might want to wrap all the cmp.Equal's with a sort option whenever comparing `[]string`'s

Comment (laurentsimon) [reply build-go/pkg/build_test.go:758]: oops, good catch. I copied from an earlier test and forgot that.

**Pull request merged**